src/internal/kopia/snapshot_manager.go

```diff
@@ -203,7 +243,24 @@ func fetchPrevManifests(
 		return nil, errors.Wrap(err, "fetching previous manifests")
 	}
 
-	return manifestsSinceLastComplete(mans), nil
+	found, hasCompleted := manifestsSinceLastComplete(mans)
+	res := make([]*ManifestEntry, 0, len(found))
+
+	for _, m := range found {
+		res = append(res, &ManifestEntry{
+			Manifest: m,
+			Reasons:  []Reason{reason},
+		})
+	}
+
+	// If we didn't find another complete manifest then we need to make the
+	// previous complete manifest as having this ResourceOwner, Service, Category
+	// as the reason as well.
+	if !hasCompleted && man != nil {
```

Comment:
Just checking; is `hasCompleted` equivalent to `lastCompleteIdx >= 0`?

Reply:
not in this case. `hasCompleted` is from `manifestsSinceLastComplete` which deal exclusively with uncached (or incomplete) manifests. `lastCompleteIdx` is the index of the last complete backup in all manifests that match the given tags, whether or not they're complete

This is checking if we did not find a complete manifest in the uncached manifests newer than the most recent cached complete manifest and that we found a cached manifest